docs/setup/keadm.md

```diff
@@ -96,70 +96,308 @@ KubeEdge edgecore is running, For logs visit:  /var/log/kubeedge/edgecore.log
 
+    Before you deploy metrics-server, you have to make sure that you deploy it on the node which has apiserver deployed on. In my case, that is my master node. As a consequence, I need to make master node schedulable by the following command:
+
+    ``` shell
+    kubectl taint nodes --all node-role.kubernetes.io/master-
+    ```
+   
+    Then, in the deployment.yaml file, I must specify that metrics-server is deployed on master node.
+    (I used hostname as the marked label.)
+    In `metrics-server-deployment.yaml`
+    ``` yaml
+        spec:
+          affinity:
+            nodeAffinity:
+              requiredDuringSchedulingIgnoredDuringExecution:
+                nodeSelectorTerms:
+                - matchExpressions:
+                  #Specify which label in [kubectl get nodes --show-labels] you want to match
+                  - key: kubernetes.io/hostname
+                    operator: In
+                    values:
+                    #Specify the value in key
+                    - charlie-latest
+    ```
+
+    The deployment YAML files can be found in this repository:
+    [DockerAndKubernetesCourseCode](https://github.com/DanWahlin/DockerAndKubernetesCourseCode/tree/master/samples/prometheus/metrics-server)
+
 **IMPORTANT NOTE:**
 1. Metrics-server needs to use hostnetwork network mode.
 
-2. Use the image compiled by yourself and set imagePullPolicy to IfNotPresent.
+2. Use the image compiled by yourself and set imagePullPolicy to Never.
 
 3. Enable the feature of --kubelet-use-node-status-port for Metrics-server
 
+    Those settings need to be written in deployment yaml (metrics-server-deployment.yaml) file like this:
+
+    ``` yaml
+          volumes:
+          # mount in tmp so we can safely use from-scratch images and/or read-only containers
+          - name: tmp-dir
+            emptyDir: {}
+          hostNetwork: true                          #Add this line to enable hostnetwork mode
+          containers:
+          - name: metrics-server
+            image: metrics-server-kubeedge:latest    #Make sure that the REPOSITORY and TAG are correct
+            # Modified args to include --kubelet-insecure-tls for Docker Desktop (don't use this flag with a real k8s cluster!!)
+            imagePullPolicy: Never                   #Make sure that the delpoyment uses the image you built up
+            args:
+              - --cert-dir=/tmp
+              - --secure-port=4443
+              - --v=2
+              - --kubelet-insecure-tls
+              - --kubelet-preferred-address-types=InternalDNS,InternalIP,ExternalIP,Hostname
+              - --kubelet-use-node-status-port       #Enable the feature of --kubelet-use-node-status-port for Metrics-server
+            ports:
+            - name: main-port
+              containerPort: 4443
+              protocol: TCP
+    ```
+
+
+    Metrics server works finally!
+
+    <img src="../images/metrics-server/final_result.png" width="713" height="376">
```

Comment:
The pic is redundant.

Reply:
I see. I will delete the picture. Thanks!